### PR TITLE
Support pull-cloud-provider-azure-e2e tests on capz

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

related: https://github.com/kubernetes/test-infra/issues/24957

### CODE REVIEW

**Comment by lzhecheng** [comment]
@chewong @khenidak Hello! Do you have time to review? Or could you please assign someone else to review?

**Comment by lzhecheng** [comment]
@nilo19 Hello, please take a look at the PR.

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml [hunk 1]
```diff
@@ -258,6 +258,46 @@ presubmits:
       testgrid-tab-name: pr-cloud-provider-azure-e2e-ccm-vmss
       description: "Runs Azure specific tests with cloud-provider-azure (https://github.com/kubernetes-sigs/cloud-provider-azure)."
       testgrid-num-columns-recent: '30'
+  - name: pull-cloud-provider-azure-e2e-capz
```
Comment (nilo19) [inline config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:261]: `pull-cloud-provider-azure-e2e` enables `MixedProtocolLBService` feature gate. Have you tested if the capz cluster enabled it as well?

Comment (lzhecheng) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:261]: I checked a cluster generated with capz and the feature gate is not enabled. I will update the code.

Comment (lzhecheng) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:261]: I added env var `TEST_CCM: "true"` for `pull-cloud-provider-azure-e2e-capz`, then this check has the same yaml file as the already existing `pull-cloud-provider-azure-e2e-ccm-capz`. Besides, according to the old aks-engine checks' config files, these 2 checks should have the same cluster config.

**Comment by nilo19** [comment]
@feiskyer could you help to approve?

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**